lib/tasks/serve.js

```diff
@@ -145,8 +145,10 @@ exports.start = function (mage, options, cb) {
 
 exports.shutdown = function (mage, options, cb) {
 	function closeServers(callback) {
-		mage.core.msgServer.close();
-		mage.core.httpServer.close(callback);
+		async.waterfall([
```

Comment:
Could you `async.series` instead? Waterfall has a very nasty habit (well, by design) to pass along arguments to the next function, which means that if you call `cb(null)` anywhere for example, your `next` will become `null` and the callback will be the 2nd argument you don't receive.